Can't install latest lfortran version

Two problems.

1. The installation instructions are misleading on what to do with a tarball after downloading it. They currently include as an example
```
tar xzf lfortran-0.42.0.tar.gz
cd lfortran-0.42.0.tar.gz
```
but I think the second line should have been
```
cd lfortran-0.42.0
```

2. I had actually replaced 0.42.0 by 0.42.0-835-ge375d9063 in the above as I wanted to see the effects of more recent changes. But the cmake command failed:
```
(lf) john:~/lfortran-0.42.0-835-ge375d9063$ cmake -DWITH_LLVM=yes -DCMAKE_INSTALL_PREFIX=`pwd`/inst . 
CMake Error at /usr/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find StaticZSTD (missing: zstd_LIBRARY zstd_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake-3.30/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindStaticZSTD.cmake:15 (find_package_handle_standard_args)
  CMakeLists.txt:151 (find_package)


-- Configuring incomplete, errors occurred!
(lf) john:~/lfortran-0.42.0-835-ge375d9063$
```
What should the instructions have said?

The `cd` command is indeed incorrect, but @harperjf I am unable to reproduce this issue on Ubuntu. Could you please try reinstalling `libzstd-dev` and see if it works?

I get a different error however, which is absent when building `lfortran-0.42.0`.

```console
(lf_release_2) saurabh-kumar@Awadh:~/Projects/System/lfortran$ cd ./lfortran-0.42.0-835-ge375d9063/
(lf_release_2) saurabh-kumar@Awadh:~/Projects/System/lfortran/lfortran-0.42.0-835-ge375d9063$ cmake -DWITH_LLVM=yes -DCMAKE_INSTALL_PREFIX=`pwd`/inst .
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.a (found version "1.3.1")
-- Found StaticZSTD: /usr/lib/x86_64-linux-gnu/libzstd.a
-- Found zstd: /usr/lib/x86_64-linux-gnu/libzstd.a
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.12.7")
-- Found LLVM 19.1.6
-- Using LLVMConfig.cmake in: /home/saurabh-kumar/miniforge3/envs/lf_release_2/lib/cmake/llvm


Configuration results
---------------------
LFORTRAN_VERSION: 0.42.0-835-ge375d9063
CPACK_PACKAGE_FILE_NAME: lfortran-0.42.0-835-ge375d9063-Linux
C compiler      : /usr/bin/cc
C++ compiler    : /usr/bin/c++
Build type: Release
C compiler flags      : -O3 -DNDEBUG
C++ compiler flags    : -Wall -Wextra -O3 -funroll-loops -DNDEBUG
Installation prefix: /home/saurabh-kumar/Projects/System/lfortran/lfortran-0.42.0-835-ge375d9063/inst
WITH_LFORTRAN_ASSERT: no
LFORTRAN_STATIC_BIN: no
LFORTRAN_BUILD_TO_WASM: no
WITH_STACKTRACE: no
WITH_RUNTIME_STACKTRACE: 
WITH_UNWIND: no
WITH_LIBUNWIND: yes
WITH_BFD: no
WITH_LLVM_STACKTRACE: no
WITH_DWARFDUMP: no
WITH_LINKH: no
WITH_MACHO: no
HAVE_LFORTRAN_DEMANGLE: yes
WITH_LLVM: yes
WITH_MLIR: no
WITH_XEUS: no
WITH_JSON: no
WITH_LSP: no
WITH_FMT: no
WITH_BENCHMARKS: no
WITH_LFORTRAN_BINARY_MODFILES: YES
WITH_RUNTIME_LIBRARY: YES
WITH_WHEREAMI: yes
WITH_ZLIB: yes
WITH_TARGET_AARCH64: no
WITH_TARGET_X86: yes
WITH_TARGET_WASM: no
WITH_KOKKOS: no
WITH_CCACHE: no
CXXFLAGS: 
CFLAGS: 
-- Configuring done (0.8s)
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
LIBUNWIND_LIBRARY
    linked by target "asr" in directory /home/saurabh-kumar/Projects/System/lfortran/lfortran-0.42.0-835-ge375d9063/src/libasr
    linked by target "lfortran_lib" in directory /home/saurabh-kumar/Projects/System/lfortran/lfortran-0.42.0-835-ge375d9063/src/lfortran

-- Generating done (0.1s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

Tried installing zstd but there is now a different show-stopping error. I tried putting the commands in a bash shell script but that caused more errors not related to Lfortran. I therefore put each one in at the command line; here is my dialogue with my Ubuntu system.
```
(lf) john:~$ conda create -n lf python cmake llvmdev
WARNING: A conda environment already exists at '/home/john/miniforge3/envs/lf'

Remove existing environment?
This will remove ALL directories contained within this specified prefix directory, including any other conda environments.

 (y/[n])? n


CondaSystemExit: Exiting.

(lf) john:~$ conda activate lf
(lf) john:~$ conda install zstd-static
Channels:
 - conda-forge
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/john/miniforge3/envs/lf

  added / updated specs:
    - zstd-static


The following NEW packages will be INSTALLED:

  zstd-static        conda-forge/linux-64::zstd-static-1.5.6-hac33072_0 


Proceed ([y]/n)? y


Downloading and Extracting Packages:

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(lf) john:~$ tar xzf lfortran-0.42.0-835-ge375d9063.tar.gz
(lf) john:~$ cd lfortran-0.42.0-835-ge375d9063
(lf) john:~/lfortran-0.42.0-835-ge375d9063$ cmake -DWITH_LLVM=yes -DCMAKE_INSTALL_PREFIX=`pwd`/inst .
-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.a (found version "1.3.1")
-- Found StaticZSTD: /home/john/miniforge3/envs/lf/lib/libzstd.a
-- Found zstd: /home/john/miniforge3/envs/lf/lib/libzstd.a
-- Found LibXml2: /usr/lib/x86_64-linux-gnu/libxml2.so (found version "2.12.7")
-- Found LLVM 19.1.6
-- Using LLVMConfig.cmake in: /home/john/miniforge3/envs/lf/lib/cmake/llvm


Configuration results
---------------------
LFORTRAN_VERSION: 0.42.0-835-ge375d9063
CPACK_PACKAGE_FILE_NAME: lfortran-0.42.0-835-ge375d9063-Linux
C compiler      : /usr/bin/cc
C++ compiler    : /usr/bin/c++
Build type: Release
C compiler flags      : -O3 -DNDEBUG
C++ compiler flags    : -Wall -Wextra -O3 -funroll-loops -DNDEBUG
Installation prefix: /home/john/lfortran-0.42.0-835-ge375d9063/inst
WITH_LFORTRAN_ASSERT: no
LFORTRAN_STATIC_BIN: no
LFORTRAN_BUILD_TO_WASM: no
WITH_STACKTRACE: no
WITH_RUNTIME_STACKTRACE: 
WITH_UNWIND: no
WITH_LIBUNWIND: yes
WITH_BFD: no
WITH_LLVM_STACKTRACE: no
WITH_DWARFDUMP: no
WITH_LINKH: no
WITH_MACHO: no
HAVE_LFORTRAN_DEMANGLE: yes
WITH_LLVM: yes
WITH_MLIR: no
WITH_XEUS: no
WITH_JSON: no
WITH_LSP: no
WITH_FMT: no
WITH_BENCHMARKS: no
WITH_LFORTRAN_BINARY_MODFILES: YES
WITH_RUNTIME_LIBRARY: YES
WITH_WHEREAMI: yes
WITH_ZLIB: yes
WITH_TARGET_AARCH64: no
WITH_TARGET_X86: yes
WITH_TARGET_WASM: no
WITH_KOKKOS: no
WITH_CCACHE: no
CXXFLAGS: 
CFLAGS: 
-- Configuring done (0.9s)
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
LIBUNWIND_LIBRARY
    linked by target "asr" in directory /home/john/lfortran-0.42.0-835-ge375d9063/src/libasr
    linked by target "lfortran_lib" in directory /home/john/lfortran-0.42.0-835-ge375d9063/src/lfortran

-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
(lf) john:~/lfortran-0.42.0-835-ge375d9063$ 
```

--- a/doc/src/installation.md
+++ b/doc/src/installation.md
@@ -49,6 +49,12 @@ First we have to install dependencies, for example using Conda:
 conda create -n lf python cmake llvmdev zstd-static zlib
 conda activate lf
 ```
+
+On a Linux system, we additionally need to install `libunwind`:
+```bash
+conda install libunwind
+```
+
 Then download a tarball from
 [https://lfortran.org/download/](https://lfortran.org/download/),
 e.g.:

--- a/doc/src/installation.md
+++ b/doc/src/installation.md
@@ -42,10 +42,11 @@ The source tarball of LFortran depends on:
 * cmake
 * LLVM 10-16 (versions 17 and 18 not supported yet)
 * zstd-static
+* zlib
 
 First we have to install dependencies, for example using Conda:
 ```bash
-conda create -n lf python cmake llvmdev zstd-static
+conda create -n lf python cmake llvmdev zstd-static zlib
 conda activate lf
 ```
 Then download a tarball from

--- a/doc/src/installation.md
+++ b/doc/src/installation.md
@@ -36,15 +36,16 @@ and selecting `New->Fortran`.
 
 This method is the recommended method if you just want to install LFortran, either yourself or in a package manager (Spack, Conda, Debian, etc.). The source tarball has all the generated files included and has minimal dependencies.
 
-The source tarball of LFortran only depends on:
+The source tarball of LFortran depends on:
 
 * Python
 * cmake
 * LLVM 10-16 (versions 17 and 18 not supported yet)
+* zstd-static
 
 First we have to install dependencies, for example using Conda:
 ```bash
-conda create -n lf python cmake llvmdev
+conda create -n lf python cmake llvmdev zstd-static
 conda activate lf
 ```
 Then download a tarball from
@@ -53,7 +54,7 @@ e.g.:
 ```bash
 wget https://github.com/lfortran/lfortran/releases/download/v0.42.0/lfortran-0.42.0.tar.gz
 tar xzf lfortran-0.42.0.tar.gz
-cd lfortran-0.42.0.tar.gz
+cd lfortran-0.42.0
 ```
 And build:
 ```

